packages/test/test-gc-sweep-tests/src/fluidObjectTracker.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { IRandom } from "@fluid-internal/stochastic-test-utils";
+import { IChannelPath } from "./handlesTracker";
+import { DataObjectManyDDSes } from "./testDataObjects";
+
+export interface IFluidObjectPath {
+	dataStoreId: string; // which DataStore/DataObject?
+	ddsId?: string; // which DDS/SharedObject?
+}
+
+export class FluidObjectTracker {
+	private readonly allFluidObjects: Map<string, IFluidObjectPath> = new Map();
+	private readonly handleChannels: Map<string, IChannelPath> = new Map();
+
+	private get usableFluidObjects(): IFluidObjectPath[] {
+		// TODO: add logic to filter out handles to unreferenced datastores that will be deleted.
+		return Array.from(this.allFluidObjects.values());
+	}
+
+	private get handleChannelPaths(): IChannelPath[] {
+		return Array.from(this.handleChannels.values());
+	}
+
+	public getRandomFluidObject(random: IRandom): IFluidObjectPath {
+		return random.pick(this.usableFluidObjects);
+	}
+
+	public getRandomHandleChannel(random: IRandom): IChannelPath {
+		return random.pick(this.handleChannelPaths);
+	}
+
+	public trackDataObject(dataObject: DataObjectManyDDSes) {
+		this.addFluidObject({ dataStoreId: dataObject.id });
+        for (const id of dataObject.getChannelIds()) {
+            this.addFluidObject({ dataStoreId: dataObject.id, ddsId: id });
+        }
+        for (const id of dataObject.getHandleChannelIds()) {
+            this.addChannel({ dataStoreId: dataObject.id, ddsId: id });
+        }
+	}
+
+	private addFluidObject(fluidObject: IFluidObjectPath) {
+		const key = fluidObject.ddsId ? `${fluidObject.dataStoreId}/${fluidObject.ddsId}` : fluidObject.dataStoreId;
```

Comment:
Would it be better to include the leading `/` on this like in the actual GC routes?  Or am I mixed up about that?

Reply:
This is just for local test storage, so it doesn't really matter what I'm storing. I'm just trying to create a path to a fluid object, the real data is stored as the IFluidObjectPath